idb/postgres/postgres.go

```diff
@@ -671,7 +674,7 @@ func buildTransactionQuery(tf idb.TransactionFilter) (query string, whereArgs []
 	}
 
 	// join in the root transaction if the returnInnerTxnOnly flag is false
-	if !tf.ReturnInnerTxnOnly {
+	if !tf.ReturnInnerTxnOnly || tf.ReturnRootTxnsOnly {
```

Comment:
is this join needed when tf.ReturnRootTxnsOnly is true? I think `t.txid IS NOT NULL` returns all the root txns already?

Reply:
at this time, when tf.ReturnInnerTxnOnly is true, it overrides tf.ReturnRootTxnsOnly setting at line 669.  is it what we want? do we return only the inner txns if we get a "bad" query where tf.ReturnInnerTxnOnly==true and tf.ReturnRootTxnsOnly==true? maybe, a valid query should only contain ReturnRootTxnsOnly or ReturnInnerTxnOnly.

Reply:
I think you're correct, we don't need the extra join here.

Reply:
Yes, I think we specifically want to avoid the join
```suggestion
	if !tf.ReturnInnerTxnOnly && !tf.ReturnRootTxnsOnly {
```